src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -174,15 +199,21 @@ public void RemoveAt(int index)
         {
             get
             {
-                if (index >= _size)
+                // Following trick can reduce the range check by one comparison
+                if ((uint)index >= (uint)Count)
```

Comment:
The issue is `Quirks with exception behavior of ReadOnlyCollectionBuilder` see https://github.com/dotnet/corefx/issues/14059 . So I think I should make such changes to produce correct Exceptions.